openfoodfoundation/openfoodnetwork
# "Enterprise Fee Summary" report

#### What? Why?

~Closes~ Relates to #2616 
~Closes~ Relates to #2656 
~Closes~ Relates to #2672 

#### What should we test?

This is best to be tested with existing OCs (if already available) that would already have data for this report. Before testing, it would be helpful to identify hubs, producers, and coordinators that would have data needed for Step 2 below.

Each of the following roles would be checked for Steps 2 and 3.

* Admin
* Hub
* Producer
* Coordinator

To test:

1. Check that HTML and CSV report generation work. These are using the same data objects, so except occasionally, for Steps 2 and 3 there is no need to check results for both formats - just checking one should be sufficient.
2. For each user role listed above, check that these data are properly summarized:
    a. Coordinator fee for order cycle
    b. Coordinator fee for incoming exchange
    c. Coordinator fee for outgoing exchange
    d. Producer fee for incoming exchange
    e. Distributor fee for outgoing exchange 
    f. Shipping fee
    g. Payment method fee
3. Do a smoke test that the correct filter options are available to each user role.
4. Test that using each filter field works. (No need to do this for each user role.)

#### Release notes

- Add "Enterprise Fee Summary" report.

Changelog Category: Added

#### Discourse thread

- https://community.openfoodnetwork.org/t/report-that-isolates-various-enterprise-fees/1361

#### Dependencies

- #3142 
- #3088 
- #3213
